astropy/io/fits/connect.py

```diff
@@ -165,6 +165,10 @@ def read_table_fits(input, hdu=None):
     for key, value, comment in table.header.cards:
 
         if key in ['COMMENT', 'HISTORY']:
+            # Convert to astropy's format
+            if key == 'COMMENT':
+                key = 'COMMENTS'
```

Comment:
Maybe @taldcroft can confirm... But I think `Table` metadata key is case-sensitive, although FITS is not.